compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -802,6 +806,47 @@ impl FirstSets {
     }
 }
 
+// Most `mbe::TokenTree`s are pre-existing in the matcher, but some are defined
+// implicitly, such as opening/closing delimiters and sequence repetition ops.
+// This type encapsulates both kinds. It implements `Clone` while avoiding the
+// need for `mbe::TokenTree` to implement `Clone`.
+#[derive(Clone, Debug)]
+enum TtHandle<'tt> {
+    Token(Token),
+    TtRef(&'tt mbe::TokenTree),
+}
+
+impl<'tt> TtHandle<'tt> {
+    fn from_token(tok: Token) -> Self {
+        TtHandle::Token(tok)
+    }
+
+    fn from_token_kind(kind: TokenKind, span: Span) -> Self {
+        TtHandle::from_token(Token::new(kind, span))
+    }
```

Comment:
Both `from_token` and `from_token_kind` looks trivial enough to me to not be an improvement.

Reply:
(This also applies to `(open,close)_token_kind`.)